File: backend/src/main/java/com/company/project/domain/port/in/CreateIncidenciaUseCase.java
package com.company.project.domain.port.in;

import com.company.project.application.command.CreateIncidenciaCommand;
import com.company.project.domain.model.Incidencia;

public interface CreateIncidenciaUseCase {
    Incidencia execute(CreateIncidenciaCommand command);
}

File: frontend/src/app/features/incidencia/application/create-incidencia.usecase.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Incidencia } from '../domain/incidencia.model';
import { INCIDENCIA_API_PORT, IncidenciaApiPort } from './port/out/incidencia-api.port';

@Injectable({ providedIn: 'root' })
export class CreateIncidenciaUseCase {
  private readonly api = inject(INCIDENCIA_API_PORT) as IncidenciaApiPort;

  execute(incidencia: Incidencia): Observable<Incidencia> {
    return this.api.create(incidencia);
  }
}
